go.mod

```diff
@@ -21,11 +21,11 @@ require (
 	sigs.k8s.io/controller-runtime v0.16.3
 	sigs.k8s.io/controller-tools v0.11.3
 	sigs.k8s.io/kustomize v2.0.3+incompatible
-	vitess.io/vitess v0.10.3-0.20240112195828-c53420174d46
+	vitess.io/vitess v0.10.3-0.20240118194106-4c36f2aa6469
```

Comment:
I upgraded the vitess dependency in order to benefit from the latest changes brought by https://github.com/vitessio/vitess/pull/14988